src/datasource.ts

```diff
@@ -577,11 +590,33 @@ async function findAssetTimeseries(
     : {
         assetIds: [assetId],
       };
-
   // since /dataquery can only have 100 items and checkboxes become difficult to use past 100 items,
   //  we only get the first 100 timeseries, and show a warning if there are too many timeseries
   const limit = 101;
-  const ts = await getTimeseries({ filter, limit }, connector);
+  let ts = [];
+  if (assetQuery.includeSubTimeseries !== false) {
+    const timeseries = await getTimeseries({ filter, limit }, connector);
+    if (!isEmpty(timeseries)) {
+      ts = concat(ts, ...timeseries);
+    }
+  }
+  if (assetQuery.withRelationships) {
+    const { labels, dataSetIds, sourceExternalIds, limit } = assetQuery.relationshipsQuery;
+    const relationshipsList = await fetchRelationships(
+      {
+        ...filterLabels(labels),
+        ...filterdataSetIds(dataSetIds),
+        ...filterExternalId(sourceExternalIds),
+        ...timeFrame,
+      },
+      limit,
+      connector
+    );
+    if (!isEmpty(relationshipsList)) {
+      ts = concat(ts, ...map(relationshipsList, 'target'));
+    }
+  }
+  ts = uniqBy(ts, 'id');
```

Comment:
could you refactor this a bit?
Maybe make two variables
```
let timeseriesFromAssets = []
let timeseriesFromRelationships = []
```
and do concat only once in the end.

something like:

```
ts = uniqBy([...timeseriesFromAssets, timeseriesFromRelationships], 'id')
```
you don't need `isEmpty` checks I believe

Reply:
```
let timeseriesFromAssets = assetQuery.includeSubTimeseries !== false ? await getTimeseries({ filter, limit }, connector) : [];
```
as simple as that?